### PR TITLE
Set email in environment when running `register_new_device`

### PR DESCRIPTION
**Pull request by mokagio**
### Description
This removes a tiny bit of annoying friction I keep bumping into when addressing internal requests to add a device to the Developer Portal.

### Testing Instructions

You could run this lane yourself with an existing device. Or, just reference the screenshot below where you can see the email input and that no email is requested afterwards. Please note that the copy and formatting are different from the final diff, which I improved.

Untitled

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: fastlane/Fastfile
Comment (mokagio) [inline fastlane/Fastfile]: "this will not be saved," as in, if you `echo $FASTLANE_USER` after the lane runs, it will return `nil`. Is that clear or should I explain more? Is it even worth clarifying?

Comment (AliSoftware) [reply fastlane/Fastfile]: That's clear enough for me (and already how I interpreted it when I read the code first before reading your GH comment) 👍

File: fastlane/Fastfile
Comment (AliSoftware) [inline fastlane/Fastfile]:
```suggestion
      UI.important("#{asc_user_email_key} environment variable not set. Set and `export` it in your `~/.zshrc` to avoid being asked about it next time.")
```

I find the "in the environment" wording a bit unclear without additional context. Is that "environment" in the sense of "the current repository" or something? So I personally prefer using the term `environment variable` to make things clearer.

File: fastlane/Fastfile
Comment (AliSoftware) [inline fastlane/Fastfile]: One drawback I see from doing the test like this is if the user defined their fastlane user in any other place:
 - Like if a username was defined in a `Matchfile` or `Appfile`
 - Or if the user used fastlane's [`CredentialManager` — whose doc can be found here](https://github.com/fastlane/fastlane/blob/e874a47c6e2e0e61590a03d3b71e75e5a505d1ce/credentials_manager/README.md) — to store the username

Which means that if I defined my username in one of those alternate ways, your lane will prompt me (`UI.input`) for my username anyway even if it's already defined by some other mean that `match` would have found and used

---

I'm sure fastlane has some methods or classes we can call to check if a username is defined (and would be used by default by `match`), wherever it might be defined (ENV, file, …), and using that would probably be a more resilient solution.

I just didn't take the time to dig deep into fastlane's codebase to find what class or method would achieve that, but maybe it could be worth taking a quick look to see if such a thing exists? [EDIT: Maybe calling [`CredentialManager.new.user`](https://github.com/fastlane/fastlane/blob/b8045bb6ab964bd334a2c87a297ed3a7f192eb4f/credentials_manager/lib/credentials_manager/account_manager.rb#L30-L35) would do the trick? 🤔 ]

(PS: if, after giving the above a try, you can't make it work in a reasonable time, I think it would still be acceptable to assume that most of us use `export FASTLANE_USER=…` in their `~/.zshrc` and that the test here would be good enough — instead of spending too much time in finding a more resilient solution. But it's maybe still worth a try to see if some quick alternative solutions would work first, before bailing before sinking into a rabbit hole?)

**Comment by mokagio** [comment]
> Maybe calling [CredentialManager.new.user](https://github.com/fastlane/fastlane/blob/b8045bb6ab964bd334a2c87a297ed3a7f192eb4f/credentials_manager/lib/credentials_manager/account_manager.rb#L30-L35) would do the trick?

Unfortunately, `CredentialsManager::AccountManager.new.user` [will prompt for the username](https://github.com/fastlane/fastlane/blob/ff41f3556f1baf8e700477e50f65be728780aa94/credentials_manager/lib/credentials_manager/account_manager.rb#L37) if it can't create a user from the environment or `Appfile`. There's no way I can see of bypassing that behavior ([unless the shell is non-interactive](https://github.com/fastlane/fastlane/blob/ff41f3556f1baf8e700477e50f65be728780aa94/credentials_manager/lib/credentials_manager/account_manager.rb#L125) but that's not the case for us).

---

Update: @AliSoftware I played around it a bit more and came up with a sort of good enough approach/hack, [here](https://github.com/woocommerce/woocommerce-ios/commit/050de3c78295f8fb8552bc872c445af47537208a).

The only thing is misses is suggesting the caller to set the env var to avoid being asked again, but it does achieve the goal of not prompting twice for the same information.

**Comment by AliSoftware** [comment]
I like the workaround you tried https://github.com/woocommerce/woocommerce-ios/commit/050de3c78295f8fb8552bc872c445af47537208a 👍  It's a bit indirect and indeed fails to provide the help message that setting `export FASTLANE_USER=…` in `~/.zshrc` would avoid the prompt, but it's more resilient to the ways people can store the username and fits the bill, so… 🤷‍♂️ 

I'm ok with either solution, so I'll let you decide depending on your preference on the pros/cons for each 🙃

**Comment by mokagio** [comment]
Thanks @AliSoftware .  I applied the `AccounManager` change, which I agree is more flexible. I just need an explicit approval now for auto-merge to kick in. Thanks!

**Review by AliSoftware — APPROVED** [review]
:shipit:

**Pull request merged**
